transformer/Transform.go

```diff
@@ -0,0 +1,228 @@
+	if err != nil {
+		log.Fatalf("ioutil.WriteFile: %v", err)
+	}
+
+	service_yml, er := os.OpenFile(formatTarget, os.O_RDWR, 0644)
+	checkError(er)
+
+	// write some text to file
+	_, err = service_yml.WriteString(string(file))
+	checkError(err)
+
+	// save changes
+	err = service_yml.Sync()
+	checkError(err)
+
+	service_yml.Close()
+
+	//format long syntax ports
+	service_yml, _ = os.Open(formatTarget)
+	defer service_yml.Close()
+
+	var lines []string
+	scanner := bufio.NewScanner(service_yml)
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+
+	//final format for ENV_VARS, CPU and PORTS
+	text := finalFormat(lines)
+
+	//write the final service.yml
+	service_yml, _ = os.Create(formatTarget)
+	service_yml, er = os.OpenFile(formatTarget, os.O_RDWR, 0644)
+
+	_, err = service_yml.WriteString(text)
+
+	checkError(err)
+
+	return true, err
+
+}
+
+func copyToServiceYML(d map[string]docker_Service, directlyTransformed bool) (map[string]common.Service, []string) {
+
+	serviceYaml := Serviceyml{
+		Services: make(map[string]common.Service),
+		Dbs:      make([]string, 0),
+	}
+	var isDB bool
+	var err error
+	var dbs []string
+
+	directlyTransformed = true
+	for k, v := range d {
+		var current_db string
+		isDB = false
+
+		if v.Image != "" {
+			current_db, isDB = checkDB(v.Image)
+		}
+		if isDB {
+			dbs = append(dbs, current_db)
+		}
+		if !isDB {
+			var longSyntaxPorts []string
+			longSyntaxPorts = v.Expose //expose and long syntax for ports dont work together..i think?
+			if len(v.Ports.ShortSyntax) > 0 {
+				for i := 0; i < len(v.Ports.ShortSyntax); i++ {
+					longSyntaxPorts = append(longSyntaxPorts, v.Ports.ShortSyntax[i])
+				}
+			} else {
+				longSyntaxPorts = []string{}
+				for i := 0; i < len(v.Ports.Port); i++ {
+
+					longSyntax := ""
+					longSyntax = "target: " + v.Ports.Port[i].Target + "\n"
+
+					if v.Ports.Port[i].Protocol == "udp" {
+						longSyntax += "udp: " + v.Ports.Port[i].Published
+						longSyntaxPorts = append(longSyntaxPorts, longSyntax)
+					} else if v.Ports.Port[i].Protocol == "tcp" {
+						reader := bufio.NewReader(os.Stdin)
+						fmt.Printf("\nYou have chosen a TCP protocol for the port published at %s - should it be mapped as HTTP, HTTPS or TCP ? : ", v.Ports.Port[i].Published)
+						var answer string //tvbot
+						answer, _ = reader.ReadString('\n')
+						answer = strings.ToUpper(answer)
+						if answer == "TCP\n" {
+							longSyntax += "tcp: " + v.Ports.Port[i].Published
+						} else if answer == "HTTP\n" {
+							longSyntax += "http: " + v.Ports.Port[i].Published
+						} else if answer == "HTTPS\n" {
+							longSyntax += "http: " + v.Ports.Port[i].Published
+						}
+						longSyntaxPorts = append(longSyntaxPorts, longSyntax)
+					}
+
+				}
+			}
+
+			var gitPath string
+			gitPath, err = common.GitRootDir("/")
+			if err != nil {
+
+			}
+			hasGit := common.HasGit(gitPath)
+
+			var gitURL, gitBranch, buildRoot string
```

Comment:
1) Shouldn't this only happen if there is no image provided?
2) @kamerblauwlicht docker-compose.yml doesn't have the concept of a git URL and branch, so this reads the current directories `.git` directory. Is that the right approach? How would the API work?